kbase-extension/static/kbase/js/widgets/function_input/parameter_input/kbaseNarrativeParameterTextInput.js

```diff
@@ -348,134 +348,170 @@ define([
-                                                    mm: key + ' - ' + d.info[10][key], info: d.info});
-                                                // allow us to show metadata match!
-                                            }
-                                        }
-                                    }
-                                }
+                tags = true;
+            }
+            console.log($input);
+            // $.fn.select2.amd.require([
+            //     'select2/utils',
+            //     'select2/dropdown',
+            //     'select2/dropdown/closeOnSelect',
+            //     'select2/dropdown/attachBody',
+            //     'select2/selection/search',
+            //     'select2/selection/single'
+            // ], function (Utils, Dropdown, CloseOnSelect, AttachBody, Search, SingleSelect) {
+            //     var SelectionAdapter = Utils.Decorate(SingleSelect, Search);
+            //
+            //     // var DropdownAdapter = Utils.Decorate(
+            //     //     Utils.Decorate(Dropdown, CloseOnSelect),
+            //     //     AttachBody
+            //     // );
+            //
+            //     $input.select2({
+            //         language: {
+            //             noResults: function() {
+            //                 return noMatchesFoundStr;
+            //             }
+            //         },
+            //         // dropdownAdapter: DropdownAdapter,
+            //         // selectionAdapter: SelectionAdapter,
+            //         tags: true,
+            //     });
+            // })
 
-                            } else {
-                                data.results.push({id: d.name, text: d.name, info: d.info});
-                            }
-                        }
-                    }
 
-                    //always allow the name if it is set as an output name, unshift it to the front...
-                    if (query.term.trim() !== "") {
-                        if (self.isOutputName && !exactMatch) {
-                            data.results.unshift({id: query.term, text: query.term});
-                        }
+            $input.select2({
+                // matcher: self.select2Matcher,
+                language: {
+                    noResults: function() {
+                        return noMatchesFoundStr;
                     }
-
-                    // paginate results
-                    var pageSize = self.options.wsObjSelectPageSize;
-                    query.callback({results: data.results.slice((query.page - 1) * pageSize, query.page * pageSize),
-                        more: data.results.length >= query.page * pageSize});
                 },
-                formatSelection: function (object, container) {
+                tags: tags,
+                placeholder: placeholder,
+                allowClear: true,
+                selectOnBlur: true,
+                // query: function (query) {
```

Comment:
I missed a couple of spots. But I think that needs to get ported over to the new version of Select2. I'll do that.